cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -156,300 +142,3 @@ def store_flow_cell_data(self, parsed_flow_cell: FlowCellDirectoryData) -> None:
-            )
-            if self.dry_run:
-                LOG.info("Dry run will not process selected project")
-                return
-            cgstats_select_parameters: List[str] = [
-                "--database",
-                self.stats_api.db_uri,
-                "select",
-                "--project",
-                project_id,
-                flow_cell_id,
-            ]
-            cgstats_process: Process = Process(binary=self.stats_api.binary)
-            with open(stdout_file.as_posix(), "w") as file:
-                with redirect_stdout(file):
-                    cgstats_process.run_command(
-                        parameters=cgstats_select_parameters, dry_run=self.dry_run
-                    )
-
-    def cgstats_lanestats(self, flow_cell_path: Path) -> None:
-        """Process lane stats using cgstats."""
-        LOG.info(
-            f"{self.stats_api.binary} --database {self.stats_api.db_uri} lanestats {flow_cell_path.as_posix()}"
-        )
-        if self.dry_run:
-            LOG.info("Dry run will not add lane stats")
-            return
-        cgstats_lane_parameters: List[str] = [
-            "--database",
-            self.stats_api.db_uri,
-            "lanestats",
-            flow_cell_path.as_posix(),
-        ]
-        cgstats_process: Process = Process(binary=self.stats_api.binary)
-        cgstats_process.run_command(parameters=cgstats_lane_parameters, dry_run=self.dry_run)
-
-    def post_process_flow_cell(self, demux_results: DemuxResults) -> None:
-        """Run all the necessary steps for post-processing a demultiplexed flow cell."""
-        if not self.dry_run:
-            Path(demux_results.flow_cell.path, DemultiplexingDirsAndFiles.DELIVERY).touch()
-            # delete cgstats
-        self.add_to_cgstats(flow_cell_path=demux_results.flow_cell.path)
-        self.cgstats_select_project(
-            flow_cell_id=demux_results.flow_cell.id, flow_cell_path=demux_results.flow_cell.path
-        )
-        self.cgstats_lanestats(flow_cell_path=demux_results.flow_cell.path)
-        self.transfer_flow_cell(
-            flow_cell_dir=demux_results.flow_cell.path, flow_cell_id=demux_results.flow_cell.id
-        )
-
-    def finish_flow_cell(
-        self, bcl_converter: str, flow_cell_name: str, flow_cell_path: Path
-    ) -> None:
-        """Post-processing flow cell."""
-        LOG.info(f"Check demultiplexed flow cell {flow_cell_name}")
-        try:
-            flow_cell: FlowCellDirectoryData = FlowCellDirectoryData(
-                flow_cell_path=flow_cell_path, bcl_converter=bcl_converter
-            )
-        except FlowCellError:
-            return
-        demux_results: DemuxResults = DemuxResults(
-            demux_dir=Path(self.demux_api.out_dir, flow_cell_name),
-            flow_cell=flow_cell,
-            bcl_converter=bcl_converter,
-        )
-        if not demux_results.flow_cell.is_hiseq_x_copy_completed():
```

Comment:
We can also remove a lot of functionality from the `FlowCellDirectoryData` class, such as this `is_hiseq_x_copy_completed`

Reply:
And the associated tests